web/api/formatters/rrd2json.c

```diff
@@ -45,12 +45,12 @@ void build_context_param_list(struct context_param **param_list, RRDSET *st)
     }
 
     RRDDIM *rd1;
-    rrdset_rdlock(st);
-
-    st->last_accessed_time = now_realtime_sec();
     (*param_list)->first_entry_t = MIN((*param_list)->first_entry_t, rrdset_first_entry_t(st));
     (*param_list)->last_entry_t  = MAX((*param_list)->last_entry_t, rrdset_last_entry_t(st));
 
+    rrdset_rdlock(st);
+    st->last_accessed_time = now_realtime_sec();
```

Comment:
Since `now_realtime_sec();` will trigger a systemcall it would be more optimal to take it out of the lock, so that the critical section is as short as possible.

Reply:
Ok I moved it above the lock!